### PR TITLE
Add Basics of Bitcoin Privacy

### PR DESCRIPTION
**Pull request by MaxHillebrand**
- Address Reuse
Problem: Easy wallet clustering
Solution: Remove used address
- Inputs and Outputs
Problem: UTXOs are not fungible
Solution: Coin labeling and manual selection
- Transaction Graph
Problem: Public transaction history
Solution: Zero Link coin joins
- Network Snooping
Problem: Clear net light clients
Solution: Full node by default & BIP158

### CODE REVIEW

**Comment by nopara73** [comment]
Good categorization.

File: docs/BitcoinPrivacy.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+## Inputs and outputs
+
+Bitcoin has an accounting model of [unspent transaction outputs [UTXO]](https://bitcoin.org/en/blockchain-guide#introduction). A transaction has inputs: the coins that are spent, and outputs: the coins that are received. The input of one transaction has to be an output of a previous transaction that is not yet spent. Each UTXO is the tip of the chain of links between inputs and outputs, all the way back to a [coin base transaction](https://en.bitcoin.it/wiki/Coinbase) that pays the miner.
+
+### Problem
+
+_**UTXOs are not fungible**_
+
+Each UTXO is a unique snow flake with a public transaction history. For example, when Alice sends a coin to Bob, than Bob does not just have any random UTXO, but he has specifically the coin that Alice has sent him. When Bob sends this coin to Charlie, than Charlie can look back two hops to see the transaction from Alice to Bob. If Bob does not want Charlie to know that the transaction between Alice and Bob happened, then Bob needs to send Charlie a different coin.
```
Comment (benthecarman) [inline docs/BitcoinPrivacy.md]: > For example, when Alice sends a coin to Bob, than Bob does not just have any random UTXO, but he has specifically the coin

For example, when Alice sends a coin to Bob, **then** Bob does not just have any random UTXO, but he has **the specific** coin

File: docs/BitcoinPrivacy.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+_**Public transaction history**_
+
+Because of the input and output model of Bitcoin, there is a chain of digital signatures all the way from the coinbase reward, to the current UTXO. This transaction history is can reveal sensitive information of the spending patterns of individuals. The receiver of a coin can look back into the transaction history of the sender. And the sender can see the future spending of the receiver.
+
+### Wasabi's Solution
+
+_**Zero Link coin joins**_
+
+In order to obfuscate the link between outputs and inputs, Wasabi uses the [Zero Link](https://github.com/nopara73/zerolink) coin join protocol. The Wasabi central coordinator cannot steal and cannot spy, he simple helps many peers to build a huge transaction, with many inputs, and many outputs. The non-private inputs can be linked to their previous transaction history. However, the equal value coin join outputs with anonymity set can not be tied to the inputs.
```
Comment (benthecarman) [inline docs/BitcoinPrivacy.md]: > However, the equal value coin join outputs with anonymity set can

However, the equal value coin join outputs with **an** anonymity set can

File: docs/BitcoinPrivacy.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+This means that when sending an anonset coin, the receiver does not know about the transaction history before the coin join. And when the receiver does a coin join himself, then the sender cannot spy on the later spending patterns.
```
Comment (benthecarman) [inline docs/BitcoinPrivacy.md]: Maybe explain here that an outsider can only guess at a 1 in anonset rate.

File: docs/BitcoinPrivacy.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+### Wasabi's Solution
+
+_**Full node by default & block filters over tor**_
+
+Wasabi checks if there is a local Tor instance installed, and if yes it uses this to onion-route all the traffic to and from the network. If there Tor is not already installed, then it is installed automatically within Wasabi. This means that per default, all network communication is secured from outside snooping and the IP address is hidden.
+
+In order to fully verify everything, running a full node is essential. If bitcoind is installed on the same computer as Wasabi, then it will automatically and per default connect to the full node. It is also possible to connect Wasabi to a remote full node on another computer by specifying the local IP address or Tor hidden service in the settings. Now Wasabi pulls the verified blocks from the full node, and it also broadcasts the transactions to the P2P network from this full node.
```
Comment (benthecarman) [inline docs/BitcoinPrivacy.md]: Maybe make "bitcoind" a link to the repo bitcoin core

**Comment by benthecarman** [comment]
Really good explanations @MaxHillebrand, added a few suggestions and a couple error fixes

**Pull request merged**